packages/transaction-controller/src/TransactionController.ts

```diff
@@ -954,6 +977,24 @@ export class TransactionController extends BaseController<
         transactionMeta,
         'TransactionController:confirmExternalTransaction - Add external transaction',
       );
+
+      if (transactionMeta.type === TransactionType.swap) {
+        updatePostTransactionBalance(transactionMeta, {
+          ethQuery: this.ethQuery,
+          getTransaction: this.getTransaction.bind(this),
+          updateTransaction: this.updateTransaction.bind(this),
+        })
+          .then(({ updatedTransactionMeta, approvalTransactionMeta }) => {
+            this.hub.emit(TransactionEvent.postTransactionBalanceUpdated, {
+              transactionMeta: updatedTransactionMeta,
+              approvalTransactionMeta,
+            });
+          })
+          .catch(() => {
+            /* istanbul ignore next */
+            log('Error while updating post transaction balance');
```

Comment:
Should we include the error in this log?

Reply:
Added in https://github.com/MetaMask/core/pull/1877/commits/09d1c8eb90ae701f99623878a9bf6d98d45f06d6